packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -345,7 +358,8 @@ RestrictedApplication.propTypes = {
   environment: PropTypes.object.isRequired,
   defaultFeatureFlags: PropTypes.object,
   render: PropTypes.func.isRequired,
-  applicationMessages: PropTypes.object.isRequired,
+  applicationMessages: PropTypes.object,
+  fetchApplicationMessages: PropTypes.func,
```

Comment:
maybe oneOf or something? I'm not sure how to make a propType be one or the other

Reply:
You can do

```js
applicationMessages: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.instanceOf(Whatever)
])
```

or any other type combination.

Reply:
yeah that makes sense 👍

Reply:
sort of weird though because applicationMessages needs to be a function, that receives a language, and returns a promise 🤔

Reply:
Yeah it's a `applicationMessages(language: Language) => Promise<ApplicationMessage>` as far as I understand.